src/documents/Priorities.md

```diff
@@ -0,0 +1,107 @@
+it involves being situated beyond an element with extremal priority). If the string is followed by <code>:class</code> for one of the supported classes, this increases the "level of infinity" that the priority
+represents to that of the class, so that it will beat any extremal priority of a lower class or an unqualified extremal priority. The currently supported extremal priority classes are <code>testing</code> and <code>authoring</code>.
+So, an element listed <code>last:authoring</code> will beat an element listed <code>last:testing</code>, which will beat any element merely listed as <code>last</code>.</td>
+<td><code>"last"</code>, <code>"last:testing"</code></td>
+</table>
+
+Note that the framework's algorithm for sorting by priority uses a [stable sorting algorithm](https://en.wikipedia.org/wiki/Sorting_algorithm#Stability) so that in the absence of any priorities, any
+original ordering of the elements will be preserved. Elements without any priority will sort close to those which have been specified with an explicit numeric priority of 0, whilst retaining any original relative ordering. 
+In the case of event listeners, this will be significant since it will respect any procedural order of addition, as well as any order
+resulting from grade merging. For options distributions, the natural ordering will be derived instead from the tree's topology (distributions which travelled a longer distance will
+be weaker than those which travelled a shorter distance), and numeric priorities should not be used.
+
+## Example of priority resolution for listeners
+
+Here is part of a grade definition from the framework's Uploader component implementation, showing a recommended use of a constraint-based priority:
+
+```javascript
+fluid.defaults("fluid.uploader.errorPanel.section", {
+    gradeNames: ["fluid.viewComponent"],
+...
+    listeners: {
+        "onCreate.bindHandlers": {
+            funcName: "fluid.uploader.errorPanel.section.bindHandlers",
+            priority: "before:refreshView"
+        },
+        "onCreate.refreshView": "{that}.refreshView"
+    }
+});
+```
+
+Amongst the two `onCreate` listeners, the framework guarantees that the `bindHandlers` listener will always be invoked before `refreshView`.
+
+## Table of supported sites for priorities
+
+The following table lists all the locations in component options where priorities are supported, together with some notes about their limitations and interpretation.
+
+<table>
+  <thead>
+    <tr>
+      <td>Options Record Entry</td>
+      <td>Polarity</td>
+      <td>Notes</td>
+    </tr>
+  </thead>
+  <tr>
+  <td><a href="InfusionEventSystem.md#registering-a-listener-to-an-event"><code>listeners</code></a></td>
+  <td>Lower numeric numbers, <code>last</code>, and <code>after</code> will fire later than others</td>
+  <td>Full support for constraints, uniquifying based on <code>namespace</code></td>
```

Comment:
What does "uniquifying" mean? If it means "adding a namespace after the before/after", then the 'no support for uniquifying' phrases below makes= no sense, because above, constraints are described as including these namespaces i.e. there is no indication that you can have before/after *without* a namespace. Regardless, please try to find some other way to phrase this besides "uniquifying".